platform-hub-web/src/app/shared/hub-api/api-pagination.factory.js

```diff
@@ -0,0 +1,55 @@
+/* eslint camelcase: 0, object-shorthand: 0 */
+
+export const apiPagination = function (_) {
+  'ngInject';
+
+  const DEFAULT_PER_PAGE = 10;
+
+  let perPage;
+
+  resetPerPage();
+
+  return {
+    setPerPage,
+    resetPerPage,
+    withPaginationParams,
+    handlePaginatedResponse
+  };
+
+  function setPerPage(num) {
+    perPage = num;
+  }
+
+  function resetPerPage() {
+    perPage = DEFAULT_PER_PAGE;
+  }
+
+  function withPaginationParams(params, page) {
+    if (page) {
+      return _.merge(
+        params,
+        {
+          per_page: perPage,
+          page: page
+        }
+      );
+    }
+
+    return params;
+  }
+
+  function handlePaginatedResponse(response) {
+    const items = response.data;
+
+    const headers = response.headers();
+
+    if (headers.total && headers['per-page']) {
+      items.pagination = {
+        total: parseInt(headers.total, 10),
+        perPage: parseInt(headers['per-page'], 10)
```

Comment:
In the `app.routes.js` it is `per_page`, should this be an underscore here too?

Reply:
It's a fair question/point – the _header_ from the API uses `per-page`, but I've made it use `per_page` in the URL for the query param. Personally I prefer using the best form for each particular context, and for query params I tend to prefer underscored names. In JavaScript the variable names are `perPage`, so we have the third form anyway.

Happy to change if there's more strong feelings towards this (though I'd rather not change the header as that's the best form for header names, so it'd be changing the query param [if the router actually allows this form]).